src/ducks/work-in-progress.js

```diff
@@ -0,0 +1,268 @@
+  const annotations = localStorage.getItem(`${userId}.${ANNOTATIONS_KEY}`);
+
+  return !!workflowId && !!subjectId && !!annotations;
+}
+
+/*  Clears any work in progress the user has.
+    Needs to be called via Redux's dispatch().
+ */
+const clear = () => {
+  return (dispatch, getState) => {
+    const userId = (getState().login.user) ? getState().login.user.id : ANONYMOUS_USER_ID;
+    localStorage.removeItem(`${userId}.${WORKFLOW_ID_KEY}`);
+    localStorage.removeItem(`${userId}.${SUBJECT_ID_KEY}`);
+    localStorage.removeItem(`${userId}.${ANNOTATIONS_KEY}`);
+    
+    //Store update
+    dispatch({ type: CLEAR_TIMESTAMP });
+    console.log('WorkInProgress.clear()');
+  }
+};
+
+/*  Saves any work in progress the user has.
+    Needs to be called via Redux's dispatch().
+ */
+const save = () => {
+  return (dispatch, getState) => {
+    const userId = (getState().login.user) ? getState().login.user.id : ANONYMOUS_USER_ID;
+    const workflowId = getState().workflow.id;
+    const subjectId = getState().subject.id;
+    const annotations = getState().annotations.annotations;
+    
+    //Sanity check: make sure we have something to load.
+    if (!workflowId || !subjectId || !annotations) {
+      console.error('WorkInProgress.save() error: nothing to save.');
+      return;
+    }
+
+    localStorage.setItem(`${userId}.${WORKFLOW_ID_KEY}`, workflowId);
+    localStorage.setItem(`${userId}.${SUBJECT_ID_KEY}`, subjectId);
+    localStorage.setItem(`${userId}.${ANNOTATIONS_KEY}`, JSON.stringify(annotations));
+    
+    //Store update
+    dispatch({ type: SET_TIMESTAMP, timestamp: new Date(Date.now()) });
+    console.log('WorkInProgress.save() success');
+  };
+};
+
+/*  Loads any work in progress the user has and TRIGGERS the necessary
+    Workflow-Subject-Classification lifecycle updates.
+    Needs to be called via Redux's dispatch().
+ */
+const load = () => {
+  return (dispatch, getState) => {
+    try {
+      const userId = (getState().login.user) ? getState().login.user.id : ANONYMOUS_USER_ID;
+      const workflowId = localStorage.getItem(`${userId}.${WORKFLOW_ID_KEY}`);
+      const subjectId = localStorage.getItem(`${userId}.${SUBJECT_ID_KEY}`);  //TODO: Check if a type conversion is required.
+      const annotations = JSON.parse(localStorage.getItem(`${userId}.${ANNOTATIONS_KEY}`));
+      
+      //Sanity check: make sure we have something to load.
+      if (!workflowId || !subjectId || !annotations) {
+        console.error('WorkInProgress.save() error: nothing to load.');
+        return;
+      }
+      
+      //First, load the Workflow, then the Subject, then the Annotations.
+      //Finally, clear all saved WIP data.
+      dispatch(fetchWorkflow(workflowId)).then(() => {
+        return dispatch(fetchSubject(subjectId));
+      }).then(() => {
+        return dispatch(loadAnnotations(annotations));
+      }).then(() => {
+        console.log('WorkInProgress.load() success');
+        return dispatch(clear());
```

Comment:
Maybe I missed this in an earlier review, but why is it necessary to clear the `localStorage` here? If a user loads their saved annotation and then closes their browser, their work will be lost. Instead of being cleared, can the work remain and just be updated with any following save? 

I